aiokafka/producer.py

```diff
@@ -496,3 +496,39 @@ def _partition(self, topic, partition, key, value,
         available = list(self._metadata.available_partitions_for_topic(topic))
         return self._partitioner(
             serialized_key, all_partitions, available)
+
+    def create_batch(self, topic, partition=None):
+        """Create and return an empty MessageBatch.
+
+        The batch is not queued for send until submission to ``send_batch``.
+        Use of both ``create_batch`` and ``send`` together is discouraged as it
+        may lead to unexpected results.
+
+        Arguments:
+            topic (str): topic where the batch will be published
+            partition (int, optional): optionally specify a partition. If not
+                set, the partition will be selected using the configured
+                'partitioner' with key and value set to None.
+
+        Returns:
+            MessageBatch: empty batch to be filled and submitted by the caller.
+        """
+        partition = self._partition(topic, partition, None, None, None, None)
+        tp = TopicPartition(topic, partition)
+        return self._message_accumulator.batch_factory(tp)
+
+    def send_batch(self, batch):
+        """Submit a MessageBatch for publication.
+
+        Use of both ``send_batch`` and ``send`` together is discouraged as it
+        may lead to unexpected results.
+
+        Arguments:
+            batch (MessageBatch): batch object to be published.
+
+        Returns:
+            asyncio.Future: object that will be set when the batch is
+                delivered.
+        """
+        log.debug("Sending batch to %s", batch.tp)
```

Comment:
We need to add an explicit flag for debug messages. This is out of the scope of this PR, just a reminder.

Reply:
Agreed.